hangman/hangman.py

```diff
@@ -1,27 +1,34 @@
 import random
-my_list = ["I", "eat", "chicken", "and", "turkey"]
-random_word = str(random.choice(my_list).upper())
-print(random_word)
+def get_target_word():
+    my_list = ["I", "eat", "chicken", "and", "turkey"]
+    random_word = str(random.choice(my_list).upper())
+    print(random_word)
+    return random_word
+random_word = get_target_word()
 
-dic_list = {}
-for letters in random_word:
-    dic_list[letters] = False
-print(dic_list)
+def convert_to_dictionary():
+    dic_list = {}
+    for letters in random_word:
+        dic_list[letters] = False
+    print(dic_list)
+    return dic_list    
+dic_list = convert_to_dictionary()
 
-word = ""
-count = 0
-while True:
-    word = input("Kindly type one letter only: ").upper()
-    index = 0
-    count += 1
-    for i in random_word:
-        if word == i:
-            dic_list[word] = True
-            print("YES in {0} and it only took you {1} tries".format(index, count))
-        index += 1 
-    if word == "!":
-        for val in dic_list:
-            print(val + ": {} ".format(dic_list[val]))
-        break
-    elif word not in dic_list:
-        print("NO! try again. Type ! to quit")
\ No newline at end of file
+def hangman(word=""):
+    count = 0
+    while True:
+        word = input("Kindly type one letter only: ").upper()
+        index = 0 
+        count += 1 
+        for i in random_word:
+            if word == i:
+                dic_list[word] = True
+                print("YES in {0} after {1} tries".format(index, count))
+            index += 1 
+        if word == "!":
+            for val in dic_list:
+                print(val + ": {} ".format(dic_list[val]))
+            break
+        elif word not in dic_list:
+            print("NO! try again. Type ! to quit")
+hangman()
```

Comment:
```suggestion
def hangman_turn(word):
        index = 0 
        count += 1 
        for i in random_word:
            if word == i:
                dic_list[word] = True
                print("YES in {0} after {1} tries".format(index, count))
            index += 1 
        if word == "!":
            for val in dic_list:
                print(val + ": {} ".format(dic_list[val]))
            break
        elif word not in dic_list:
            print("NO! try again. Type ! to quit")

count = 0
while True:
    word = input("Kindly type one letter only: ").upper()
    hangman_turn()
```

Reply:
Actally, ignore this because you would still need to handle the `count` variable. There are other ways to refactor this but lets leave it has you currently have it.

Reply:
i tried editing this but if the loop happens outside the function, it won't end even when the player quits

Reply:
def hangman(word, count):
        index = 0 
        for i in random_word:
            if word == i:
                dic_list[word] = True
                print("YES in {0} after {1} tries".format(index, count))
            index += 1 
        if word == "!":
            for val in dic_list:
                print(val + ": {} ".format(dic_list[val]))
            # break
        elif word not in dic_list:
            print("NO! try again. Type ! to quit")

count = 0
while True:
    word = input("Kindly type one letter only: ").upper()
    count += 1 
    hangman(word,count)